### ISSUE TITLE
grpc install error Python 3.11.1

### ISSUE THREAD

**Issue by samgostomski** [grpc/grpc#32105]
### What version of gRPC and what language are you using?
Python == 3.11.1
gprcio ==  1.51.1

### What operating system (Linux, Windows,...) and version?
MacOS Ventura Version 13.1 (22C65)
Apple M2

### What runtime / compiler are you using (e.g. python version or version of gcc)
Python version == 3.11.1

### What did you do?
Overview: I was attempting to install the necessary packages to set up a [BigQuery python cloud client](https://cloud.google.com/python/docs/reference) and am blocked by a gRPC install error. 

Ran the following commands:

1.  `gcloud auth application-default login ` (successful)
2. `pip3 install google-cloud-runtimeconfig` (successful)
3. `pip3 install google-cloud-bigquery` (returned error, see details below)

### What did you expect to see?
Running `pip3 install google-cloud-bigquery` should have completed the installation process for the packages necessary to set up a BigQuery client. 

### What did you see instead?

Error code summary: `
  Running setup.py install for grpcio ... error
  error: subprocess-exited-with-error
`
Error long form: 

> [user]~ % pip3 install google-cloud-bigquery
Collecting google-cloud-bigquery
  Using cached google_cloud_bigquery-3.4.1-py2.py3-none-any.whl (215 kB)
Collecting grpcio<2.0dev,>=1.47.0
  Using cached grpcio-1.51.1.tar.gz (22.1 MB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-cloud-bigquery) (2.11.0)
Requirement already satisfied: proto-plus<2.0.0dev,>=1.15.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-cloud-bigquery) (1.22.2)
Requirement already satisfied: google-cloud-core<3.0.0dev,>=1.4.1 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-cloud-bigquery) (2.3.2)
Collecting google-resumable-media<3.0dev,>=0.6.0
  Using cached google_resumable_media-2.4.0-py2.py3-none-any.whl (77 kB)
Requirement already satisfied: packaging<22.0.0dev,>=14.3 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-cloud-bigquery) (21.3)
Requirement already satisfied: protobuf!=3.20.0,!=3.20.1,!=4.21.0,!=4.21.1,!=4.21.2,!=4.21.3,!=4.21.4,!=4.21.5,<5.0.0dev,>=3.19.5 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-cloud-bigquery) (4.21.12)
Requirement already satisfied: python-dateutil<3.0dev,>=2.7.2 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-cloud-bigquery) (2.8.2)
Requirement already satisfied: requests<3.0.0dev,>=2.21.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-cloud-bigquery) (2.28.2)
Requirement already satisfied: googleapis-common-protos<2.0dev,>=1.56.2 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->google-cloud-bigquery) (1.58.0)
Requirement already satisfied: google-auth<3.0dev,>=2.14.1 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->google-cloud-bigquery) (2.16.0)
Collecting grpcio-status<2.0dev,>=1.33.2
  Using cached grpcio_status-1.51.1-py3-none-any.whl (5.1 kB)
Collecting google-crc32c<2.0dev,>=1.0
  Using cached google_crc32c-1.5.0-cp311-cp311-macosx_10_9_universal2.whl (32 kB)
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from packaging<22.0.0dev,>=14.3->google-cloud-bigquery) (3.0.9)
Requirement already satisfied: six>=1.5 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from python-dateutil<3.0dev,>=2.7.2->google-cloud-bigquery) (1.16.0)
Requirement already satisfied: charset-normalizer<4,>=2 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from requests<3.0.0dev,>=2.21.0->google-cloud-bigquery) (3.0.1)
Requirement already satisfied: idna<4,>=2.5 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from requests<3.0.0dev,>=2.21.0->google-cloud-bigquery) (3.4)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from requests<3.0.0dev,>=2.21.0->google-cloud-bigquery) (1.25.11)
Requirement already satisfied: certifi>=2017.4.17 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from requests<3.0.0dev,>=2.21.0->google-cloud-bigquery) (2022.12.7)
Requirement already satisfied: cachetools<6.0,>=2.0.0 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-auth<3.0dev,>=2.14.1->google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->google-cloud-bigquery) (5.2.1)
Requirement already satisfied: pyasn1-modules>=0.2.1 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-auth<3.0dev,>=2.14.1->google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->google-cloud-bigquery) (0.2.8)
Requirement already satisfied: rsa<5,>=3.1.4 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from google-auth<3.0dev,>=2.14.1->google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->google-cloud-bigquery) (4.0)
Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages (from pyasn1-modules>=0.2.1->google-auth<3.0dev,>=2.14.1->google-api-core[grpc]!=2.0.*,!=2.1.*,!=2.2.*,!=2.3.0,<3.0.0dev,>=1.31.5->google-cloud-bigquery) (0.4.8)
Installing collected packages: grpcio, google-crc32c, grpcio-status, google-resumable-media, google-cloud-bigquery
  DEPRECATION: grpcio is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for grpcio ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for grpcio did not run successfully.
  │ exit code: 1
  ╰─> [684 lines of output]
      ASM Builds for BoringSSL currently not supported on: macosx-10.9-universal2
      Found cython-generated files...
      running install
      /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(

     (...) 

>       xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
>       
>       Could not find <Python.h>. This could mean the following:
>         * You're on Ubuntu and haven't run `apt-get install python3-dev`.
>         * You're on RHEL/Fedora and haven't run `yum install python3-devel` or
>           `dnf install python3-devel` (make sure you also have redhat-rpm-config
>           installed)
>         * You're on Mac OS X and the usual Python framework was somehow corrupted
>           (check your environment variables or try re-installing?)
>         * You're on Windows and your Python installation was somehow corrupted
>           (check your environment variables or try re-installing?)
>       Traceback (most recent call last):
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/unixccompiler.py", line 186, in _compile
>           self.spawn(compiler_so + cc_args + [src, '-o', obj] + extra_postargs)
>         File "/private/var/folders/0z/0wn4zws114s024_j8b6sq5180000gp/T/pip-install-5ltx3sss/grpcio_d7e10fd7c0f94567b59d392637d3230e/src/python/grpcio/_spawn_patch.py", line 54, in _commandfile_spawn
>           _classic_spawn(self, command)
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/ccompiler.py", line 1007, in spawn
>           spawn(cmd, dry_run=self.dry_run, **kwargs)
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/spawn.py", line 70, in spawn
>           raise DistutilsExecError(
>       distutils.errors.DistutilsExecError: command '/usr/bin/clang' failed with exit code 1
>       
>       During handling of the above exception, another exception occurred:
>       
>       Traceback (most recent call last):
>         File "/private/var/folders/0z/0wn4zws114s024_j8b6sq5180000gp/T/pip-install-5ltx3sss/grpcio_d7e10fd7c0f94567b59d392637d3230e/src/python/grpcio/commands.py", line 280, in build_extensions
>           build_ext.build_ext.build_extensions(self)
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 468, in build_extensions
>           self._build_extensions_serial()
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 494, in _build_extensions_serial
>           self.build_extension(ext)
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/command/build_ext.py", line 246, in build_extension
>           _build_ext.build_extension(self, ext)
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 549, in build_extension
>           objects = self.compiler.compile(
>                     ^^^^^^^^^^^^^^^^^^^^^^
>         File "/private/var/folders/0z/0wn4zws114s024_j8b6sq5180000gp/T/pip-install-5ltx3sss/grpcio_d7e10fd7c0f94567b59d392637d3230e/src/python/grpcio/_parallel_compile_patch.py", line 58, in _parallel_compile
>           multiprocessing.pool.ThreadPool(BUILD_EXT_COMPILER_JOBS).map(
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/pool.py", line 367, in map
>           return self._map_async(func, iterable, mapstar, chunksize).get()
>                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/pool.py", line 774, in get
>           raise self._value
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/pool.py", line 125, in worker
>           result = (True, func(*args, **kwds))
>                           ^^^^^^^^^^^^^^^^^^^
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/pool.py", line 48, in mapstar
>           return list(map(*args))
>                  ^^^^^^^^^^^^^^^^
>         File "/private/var/folders/0z/0wn4zws114s024_j8b6sq5180000gp/T/pip-install-5ltx3sss/grpcio_d7e10fd7c0f94567b59d392637d3230e/src/python/grpcio/_parallel_compile_patch.py", line 54, in _compile_single_file
>           self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/unixccompiler.py", line 188, in _compile
>           raise CompileError(msg)
>       distutils.errors.CompileError: command '/usr/bin/clang' failed with exit code 1
>       
>       During handling of the above exception, another exception occurred:
>       
>       Traceback (most recent call last):
>         File "<string>", line 2, in <module>
>         File "<pip-setuptools-caller>", line 34, in <module>
>         File "/private/var/folders/0z/0wn4zws114s024_j8b6sq5180000gp/T/pip-install-5ltx3sss/grpcio_d7e10fd7c0f94567b59d392637d3230e/setup.py", line 535, in <module>
>           setuptools.setup(
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/__init__.py", line 87, in setup
>           return distutils.core.setup(**attrs)
>                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
>           return run_commands(dist)
>                  ^^^^^^^^^^^^^^^^^^
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
>           dist.run_commands()
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
>           self.run_command(cmd)
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/dist.py", line 1208, in run_command
>           super().run_command(command)
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
>           cmd_obj.run()
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/command/install.py", line 68, in run
>           return orig.install.run(self)
>                  ^^^^^^^^^^^^^^^^^^^^^^
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/command/install.py", line 698, in run
>           self.run_command('build')
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
>           self.distribution.run_command(command)
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/dist.py", line 1208, in run_command
>           super().run_command(command)
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
>           cmd_obj.run()
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/command/build.py", line 132, in run
>           self.run_command(cmd_name)
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
>           self.distribution.run_command(command)
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/dist.py", line 1208, in run_command
>           super().run_command(command)
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
>           cmd_obj.run()
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/command/build_ext.py", line 84, in run
>           _build_ext.run(self)
>         File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/_distutils/command/build_ext.py", line 346, in run
>           self.build_extensions()
>         File "/private/var/folders/0z/0wn4zws114s024_j8b6sq5180000gp/T/pip-install-5ltx3sss/grpcio_d7e10fd7c0f94567b59d392637d3230e/src/python/grpcio/commands.py", line 283, in build_extensions
>           support.diagnose_build_ext_error(self, error, formatted_exception)
>         File "/private/var/folders/0z/0wn4zws114s024_j8b6sq5180000gp/T/pip-install-5ltx3sss/grpcio_d7e10fd7c0f94567b59d392637d3230e/src/python/grpcio/support.py", line 117, in diagnose_build_ext_error
>           diagnostic(build_ext, error)
>         File "/private/var/folders/0z/0wn4zws114s024_j8b6sq5180000gp/T/pip-install-5ltx3sss/grpcio_d7e10fd7c0f94567b59d392637d3230e/src/python/grpcio/support.py", line 76, in diagnose_compile_error
>           _expect_compile(build_ext.compiler, c_check, message)
>         File "/private/var/folders/0z/0wn4zws114s024_j8b6sq5180000gp/T/pip-install-5ltx3sss/grpcio_d7e10fd7c0f94567b59d392637d3230e/src/python/grpcio/support.py", line 68, in _expect_compile
>           raise commands.CommandError(
>       commands.CommandError: Diagnostics found a compilation environment issue:
>       
>       Could not find <Python.h>. This could mean the following:
>         * You're on Ubuntu and haven't run `apt-get install python3-dev`.
>         * You're on RHEL/Fedora and haven't run `yum install python3-devel` or
>           `dnf install python3-devel` (make sure you also have redhat-rpm-config
>           installed)
>         * You're on Mac OS X and the usual Python framework was somehow corrupted
>           (check your environment variables or try re-installing?)
>         * You're on Windows and your Python installation was somehow corrupted
>           (check your environment variables or try re-installing?)
>       
>       [end of output]
>   
> 

>   note: This error originates from a subprocess, and is likely not a problem with pip.
> error: legacy-install-failure
> 
> × Encountered error while trying to install package.
> ╰─> grpcio
> 
> note: This is an issue with the package mentioned above, not pip.
> hint: See above for output from the failure.

### Attempted Troubleshooting

I tried the following commands to work around this error, based on [this Stack Overflow article](https://stackoverflow.com/questions/66640705/how-can-i-install-grpcio-on-an-apple-m1-silicon-laptop) and another issue reported by @ zhouyq9001 here: Python Install Error #32043 

`pip3` install --no-binary :all: grpcio --ignore-installed' 

> DEPRECATION: --no-binary currently disables reading from the cache of locally built wheels. In the future --no-binary will not influence the wheel cache. pip 23.1 will enforce this behaviour change. A possible replacement is to use the --no-cache-dir option. You can use the flag --use-feature=no-binary-enable-wheel-cache to test the upcoming behaviour. Discussion can be found at https://github.com/pypa/pip/issues/11453
> Collecting grpcio
>   Using cached grpcio-1.51.1.tar.gz (22.1 MB)
>   Preparing metadata (setup.py) ... done
> Installing collected packages: grpcio
>   DEPRECATION: grpcio is being installed using the legacy 'setup.py install' method, because the '--no-binary' option was enabled for it and this currently disables local wheel building for projects that don't have a 'pyproject.toml' file. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/11451
>   Running setup.py install for grpcio ... error
>   error: subprocess-exited-with-error
>   
>   × Running setup.py install for grpcio did not run successfully.
>   │ exit code: 1
>   ╰─> [684 lines of output]
>       ASM Builds for BoringSSL currently not supported on: macosx-10.9-universal2
>       Found cython-generated files...
>       running install
>       /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
>         warnings.warn(

.... I can post the remainder of the log if helpful. 

I also tried:

`pip3 install --no-binary :all: grpcio-tools --ignore-installed` 

Resulting in a different, but still unsuccessful, output: 

```
DEPRECATION: --no-binary currently disables reading from the cache of locally built wheels. In the future --no-binary will not influence the wheel cache. pip 23.1 will enforce this behaviour change. A possible replacement is to use the --no-cache-dir option. You can use the flag --use-feature=no-binary-enable-wheel-cache to test the upcoming behaviour. Discussion can be found at https://github.com/pypa/pip/issues/11453
Collecting grpcio-tools
  Downloading grpcio-tools-1.51.1.tar.gz (2.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.3/2.3 MB 17.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting protobuf<5.0dev,>=4.21.6
  Downloading protobuf-4.21.12.tar.gz (220 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 220.3/220.3 kB 20.8 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting grpcio>=1.51.1
  Using cached grpcio-1.51.1.tar.gz (22.1 MB)
  Preparing metadata (setup.py) ... done
Collecting setuptools
  Downloading setuptools-65.7.0.tar.gz (2.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.6/2.6 MB 41.1 MB/s eta 0:00:00
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: setuptools
  Building wheel for setuptools (pyproject.toml) ... done
  Created wheel for setuptools: filename=setuptools-65.7.0-py3-none-any.whl size=1234046 sha256=4736259a9c0123a2215526ee7d194d752ceb7c0613ba9cfc74acd270282b5911
  Stored in directory: /Users/Work/Library/Caches/pip/wheels/4c/9f/ed/8802c82bd96bf4e259f857aa6ac91525d045611e2e3b0b3ead
Successfully built setuptools
Installing collected packages: setuptools, protobuf, grpcio, grpcio-tools
  DEPRECATION: protobuf is being installed using the legacy 'setup.py install' method, because the '--no-binary' option was enabled for it and this currently disables local wheel building for projects that don't have a 'pyproject.toml' file. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/11451
  Running setup.py install for protobuf ... done
  DEPRECATION: grpcio is being installed using the legacy 'setup.py install' method, because the '--no-binary' option was enabled for it and this currently disables local wheel building for projects that don't have a 'pyproject.toml' file. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/11451
  Running setup.py install for grpcio ... error
  error: subprocess-exited-with-error
```

Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

See [TROUBLESHOOTING.md](https://github.com/grpc/grpc/blob/master/TROUBLESHOOTING.md) for how to diagnose problems better.

### Anything else we should know about your project / environment?
It is worth noting that in the documentation for [Python Clients for Google BigQuery](https://cloud.google.com/python/docs/reference/bigquery/latest) lists the supported versions as Python >= 3.7, < 3.11

I did not install this in a virtual environment

**Comment by XuanWang-Amos** [comment]
You need install Python header first, but with gRPC 1.52 release (In next week) you'll be able to install prebuilt directly.

**Comment by kunjolee** [comment]
I have this issue too. Has someone got any answer?

**Comment by gnossen** [comment]
If the headers are properly installed, then running the following command in the same environment (e.g. virtualenv) as your install should print the headers:

```
python3 -c "import sysconfig; print(sysconfig.get_path('include'))"
```

If not, then there is indeed likely an issue with your Python development install.

**Comment by password36** [comment]
> 

thx,use grpcio 1.52.0 resvoled my problem
ENV mac m1 ventura 13.0.1
